File: api/Models/Inventory.cs
namespace api.Models
{
    public class Inventory
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
        public int Quantity { get; set; }
        public string Unit { get; set; } = string.Empty;
        public string Location { get; set; } = string.Empty;
        public DateTime PurchaseDate { get; set; }
        public DateTime? ExpiryDate { get; set; }
        
        public ItemStatus Status
        {
            get
            {
                if (!ExpiryDate.HasValue)
                    return ItemStatus.FRESH;

                var daysUntilExpiry = (ExpiryDate.Value.Date - DateTime.UtcNow.Date).Days;

                if (daysUntilExpiry < 0)
                    return ItemStatus.EXPIRED;
                else if (daysUntilExpiry <= 7)
                    return ItemStatus.EXPIRING_SOON;
                else
                    return ItemStatus.FRESH;
            }
        }
    }
}

File: ui/src/types/inventory.ts
export const ItemStatus = {
  FRESH: 0,
  EXPIRING_SOON: 1,
  EXPIRED: 2
} as const;

export type ItemStatus = typeof ItemStatus[keyof typeof ItemStatus];

export interface InventoryItem {
  id: string;
  name: string;
  category: string;
  quantity: number;
  unit: string;
  location: string;
  expiryDate?: string;
  status: ItemStatus;
  daysUntilExpiry?: number;
}

export interface CreateInventoryItem {
  name: string;
  category: string;
  quantity: number;
  unit: string;
  location: string;
  expiryDate?: string;
  purchaseDate: string;
}
